include/boost/json/monotonic_resource.hpp

```diff
@@ -13,30 +14,88 @@
 #include <boost/json/detail/config.hpp>
 #include <boost/json/storage_ptr.hpp>
 
+#include <limits>
+
 namespace boost {
 namespace json {
 
 /** A fast memory resource using many large fixed size blocks.
 */
 class monotonic_resource final
     : public memory_resource
-{
-    struct block;
-    std::size_t const block_size_ = 64 * 1024;
+{   
+    struct initial_block
+    {
+        std::size_t size;
+        void* buffer = nullptr;
+        std::uintptr_t top;
+
+        initial_block() = default;
+
+        initial_block(
+            void* buf,
+            std::size_t n)
+            : size(n),
+              buffer(buf),
+              top(base())
+        {
+        }
+
+        std::uintptr_t
+        base() const
+        {
+            return reinterpret_cast<
+                std::uintptr_t>(buffer);
+        }
+    };
+public:
+    struct alignas(alignof(std::max_align_t)) block;
+
+    std::size_t block_size_;
     block* head_ = nullptr;
+    initial_block initial_;
+    
+    static const std::size_t min_block_size_ = 1024;
+    static const std::size_t max_block_size_ = 
+        std::numeric_limits<std::size_t>::max();
 
-    inline block& alloc_block(std::size_t size);
+    std::uintptr_t
+    align_address(
+        std::uintptr_t ptr,
+        std::size_t align);
+
+    template<typename Block>
+    void*
+    allocate_in_block(
+        Block& blk,
+        std::size_t n,
+        std::size_t align);
+
+    block&
+    allocate_new_block(std::size_t size);
```

Comment:
this needs to be marked `inline` no?

Reply:
Ah, forgot about header only. Should just be inline though since these aren't part of the interface, right?